packages/main/src/MultiInput.ts

```diff
@@ -115,12 +117,38 @@ class MultiInput extends Input {
 	 * Defines the component tokens.
 	 * @public
 	 */
-	@slot()
+	@slot({ type: HTMLElement })
 	tokens!: Array<IToken>;
 
 	_skipOpenSuggestions: boolean;
 	_valueHelpIconPressed: boolean;
 
+	get formValidity(): ValidityStateFlags {
+		const tokens = (this.tokens || []);
+
+		return { valueMissing: this.required && !this.value && !tokens.length };
+	}
+
+	get formFormattedValue(): FormData | string | null {
+		const tokens = (this.tokens || []);
+
+		if (tokens.length) {
+			const formData = new FormData();
+
+			for (let i = 0; i < tokens.length; i++) {
+				formData.append(this.name, tokens[i].text);
+			}
+
+			if (this.value) {
+				formData.append(this.name, this.value);
```

Comment:
same as combo box, user typed value should be first (event if empty) and it should be documented